sdk/core/azure-core/azure/core/tracing/_inference_api_instrumentor_impl.py

```diff
@@ -0,0 +1,379 @@
+    if isinstance(result, _models.ChatCompletions):
+        _add_response_chat_attributes(span, result)
+        _add_response_chat_message_event(span, result)
+    # TODO add more models here
+
+
+def _accumulate_response(item, accumulate: dict[str, Any]) -> None:
+    if item.finish_reason:
+        accumulate["finish_reason"] = item.finish_reason
+    if item.index:
+        accumulate["index"] = item.index
+    if item.delta.content:
+        accumulate.setdefault("message", {})
+        accumulate["message"].setdefault("content", "")
+        accumulate["message"]["content"] += item.delta.content
+    if item.delta.tool_calls:
+        accumulate.setdefault("message", {})
+        accumulate["message"].setdefault("tool_calls", [])
+        for tool_call in item.delta.tool_calls:
+            if tool_call.id:
+                accumulate["message"]["tool_calls"].append({"id": tool_call.id, "type": "", "function": {"name": "", "arguments": ""}})
+            if tool_call.type:
+                accumulate["message"]["tool_calls"][-1]["type"] = tool_call.type
+            if tool_call.function and tool_call.function.name:
+                accumulate["message"]["tool_calls"][-1]["function"]["name"] = tool_call.function.name
+            if tool_call.function and tool_call.function.arguments:
+                accumulate["message"]["tool_calls"][-1]["function"]["arguments"] += tool_call.function.arguments
+
+
+def _wrapped_stream(stream_obj: _models.StreamingChatCompletions, span: AbstractSpan) ->  _models.StreamingChatCompletions:
+    class StreamWrapper(_models.StreamingChatCompletions):
+        def __init__(self, stream_obj):
+            super().__init__(stream_obj._response)
+
+        def __iter__(self) -> Iterator[_models.StreamingChatCompletionsUpdate]:
+            global _trace_inference_content
+            try:
+                accumulate: dict[str, Any] = {}
+                for chunk in stream_obj:
+                    for item in chunk.choices:
+                        _accumulate_response(item, accumulate)
+                    yield chunk
+
+                if _trace_inference_content:
+                    span.span_instance.add_event(
+                        name="gen_ai.choice",
+                        attributes={
+                            "get_ai.system": INFERENCE_GEN_AI_SYSTEM_NAME,
+                            "gen_ai.event.content": json.dumps(accumulate)
+                        }
+                    )
+                _add_response_chat_attributes(span, chunk)
+
+            except Exception as exc:
+                _set_attributes(span, ("error.type", exc.__class__.__name__))
+                raise
+
+            finally:
+                if stream_obj._done is False:
+                    if accumulate.get("finish_reason") is None:
+                        accumulate["finish_reason"] = "error"
+                    if _trace_inference_content:
+                        span.span_instance.add_event(
```

Comment:
there was some back and forth on how to capture events and we don't have a conclusion yet - https://github.com/open-telemetry/semantic-conventions/pull/980. 

Let's keep doing what you're doing now, but I'll probably come back later and change it to comply with the version that'll get merged.